cmd/query/app/apiv3/grpc_handler.go

```diff
@@ -32,21 +33,34 @@ func (h *Handler) GetTrace(request *api_v3.GetTraceRequest, stream api_v3.QueryS
 		return fmt.Errorf("malform trace ID: %w", err)
 	}
 
-	query := querysvc.GetTraceParameters{
-		GetTraceParameters: spanstore.GetTraceParameters{
-			TraceID:   traceID,
-			StartTime: request.GetStartTime(),
-			EndTime:   request.GetEndTime(),
+	query := querysvc.GetTraceParams{
+		TraceIDs: []tracestore.GetTraceParams{
+			{
+				TraceID: traceID.ToOTELTraceID(),
+				Start:   request.GetStartTime(),
+				End:     request.GetEndTime(),
+			},
 		},
 		RawTraces: request.GetRawTraces(),
 	}
-	trace, err := h.QueryService.GetTrace(stream.Context(), query)
-	if err != nil {
-		return fmt.Errorf("cannot retrieve trace: %w", err)
-	}
-	td := modelToOTLP(trace.GetSpans())
-	tracesData := api_v3.TracesData(td)
-	return stream.Send(&tracesData)
+	getTracesIter := h.QueryService.GetTraces(stream.Context(), query)
+	var getTraceErr error
+	getTracesIter(func(traces []ptrace.Traces, err error) bool {
+		if err != nil {
+			getTraceErr = err
+			return false
+		}
+		for _, trace := range traces {
+			tracesData := api_v3.TracesData(trace)
+			if err := stream.Send(&tracesData); err != nil {
+				getTraceErr = status.Error(codes.Internal,
+					fmt.Sprintf("failed to send response stream chunk to client: %v", err))
+				return false
+			}
+		}
+		return false // only one trace
```

Comment:
we din't know how the original trace is chunked
```suggestion
		return false // only one trace
```